File: visclient/PumpStation.cpp
// -*-c++-*-

#include "PumpStation.h"
#include "game.h"


namespace client
{

PumpStation::PumpStation(_PumpStation* pointer)
{
    ptr = (void*) pointer;
}

int PumpStation::id()
{
  return ((_PumpStation*)ptr)->id;
}

int PumpStation::owner()
{
  return ((_PumpStation*)ptr)->owner;
}

int PumpStation::waterAmount()
{
  return ((_PumpStation*)ptr)->waterAmount;
}

int PumpStation::siegeAmount()
{
  return ((_PumpStation*)ptr)->siegeAmount;
}




std::ostream& operator<<(std::ostream& stream,PumpStation ob)
{
  stream << "id: " << ((_PumpStation*)ob.ptr)->id  <<'\n';
  stream << "owner: " << ((_PumpStation*)ob.ptr)->owner  <<'\n';
  stream << "waterAmount: " << ((_PumpStation*)ob.ptr)->waterAmount  <<'\n';
  stream << "siegeAmount: " << ((_PumpStation*)ob.ptr)->siegeAmount  <<'\n';
  return stream;
}

}

File: shellAI/java/PumpStation.java
import com.sun.jna.Pointer;

///Represents a base to which you want to lead water.
class PumpStation
{
  Pointer ptr;
  int ID;
  int iteration;
  public PumpStation(Pointer p)
  {
    ptr = p;
    ID = Client.INSTANCE.pumpStationGetId(ptr);
    iteration = BaseAI.iteration;
  }
  boolean validify()
  {
    if(iteration == BaseAI.iteration) return true;
    for(int i = 0; i < BaseAI.pumpStations.length; i++)
    {
      if(BaseAI.pumpStations[i].ID == ID)
      {
        ptr = BaseAI.pumpStations[i].ptr;
        iteration = BaseAI.iteration;
        return true;
      }
    }
    throw new ExistentialError();
  }

    //commands


    //getters

  ///Unique Identifier
  public int getId()
  {
    validify();
    return Client.INSTANCE.pumpStationGetId(ptr);
  }
  ///The owner of the PumpStation.
  public int getOwner()
  {
    validify();
    return Client.INSTANCE.pumpStationGetOwner(ptr);
  }
  ///The amount the PumpStation has been sieged.
  public int getSiegeAmount()
  {
    validify();
    return Client.INSTANCE.pumpStationGetSiegeAmount(ptr);
  }

}
